File: nodejs/scripts/get_businesses.js
#!/usr/bin/env node
import {ArgumentParser} from 'argparse'

import {ApiConfig} from '../src/api_config.js'
import {common_arguments} from '../src/arguments.js'

async function main (argv) {
  const parser = new ArgumentParser({description: 'Get Pinterest OAuth token'});
  common_arguments(parser);
  const args = parser.parse_args(argv);

  // get configuration from defaults and/or the environment
  const api_config = new ApiConfig({verbosity: args.log_level, version: args.api_version});

  const {AccessToken} = await import(`../src/${api_config.version}/access_token.js`);
  const {Scope} = await import(`../src/${api_config.version}/oauth_scope.js`);
  const {User} = await import(`../src/${api_config.version}/user.js`);

  // Note that the OAuth will fail if your application does not
  // have access to the scope that is required to access
  // linked business accounts.
  const access_token = new AccessToken(api_config, {name: args.access_token});
  await access_token.fetch({scopes: [Scope.READ_USERS, Scope.READ_ADVERTISERS]});

  // use the access token to get information about the user
  const user_me = new User('me', api_config, access_token);
  const user_me_data = await user_me.get();
  user_me.print_summary(user_me_data);

  console.log('trying to get businesses...')
  const user_me_businesses = await user_me.get_businesses();
  if (user_me_businesses && (user_me_businesses.length != 0)) {
    console.log(user_me_businesses);
  } else {
    console.log('This account has no information on linked businesses.');
  }
}

if (!process.env.TEST_ENV) {
  main(process.argv.slice(2));
}

File: python/scripts/get_businesses.py
#!/usr/bin/env python
import argparse
import sys
from os.path import abspath, dirname, join

sys.path.append(abspath(join(dirname(__file__), "..", "src")))

from access_token import AccessToken
from api_config import ApiConfig
from arguments import common_arguments
from oauth_scope import Scope
from user import User


def main(argv=[]):
    """
    This script gets the business accounts associated with a User.

    Strictly speaking, the READ_ADVERTISERS scope is neither
    necessary nor sufficient for this script to execute.
    However, obtaining the READ_ADVERTISERS scope is a good way to
    verify that the application is authorized to request the
    /v3/users/me/businesses endpoint.
    """
    parser = argparse.ArgumentParser(description="Get User Businesses")
    common_arguments(parser)
    args = parser.parse_args(argv)

    # get configuration from defaults and/or the environment
    api_config = ApiConfig(verbosity=args.log_level, version=args.api_version)

    # Note that the OAuth will fail if your application does not
    # have access to the scope that is required to access
    # linked business accounts.
    access_token = AccessToken(api_config, name=args.access_token)
    access_token.fetch(scopes=[Scope.READ_USERS, Scope.READ_ADVERTISERS])

    # use the access token to get information about the user
    user_me = User("me", api_config, access_token)
    user_me_data = user_me.get()
    user_me.print_summary(user_me_data)

    print("trying to get businesses...")
    user_me_businesses = user_me.get_businesses()
    if user_me_businesses:
        print(user_me_businesses)
    else:
        print("This account has no information on linked businesses.")


if __name__ == "__main__":
    main(sys.argv[1:])
